runtime/src/accounts.rs

```diff
@@ -141,8 +142,15 @@ impl Accounts {
                 error_counters.insufficient_funds += 1;
                 Err(TransactionError::InsufficientFundsForFee)
             } else {
-                accounts[0].lamports -= fee;
-                Ok((accounts, rents))
+                match hash_age_kind {
+                    HashAgeKind::FullSpend if accounts[0].lamports != fee => {
+                        Err(TransactionError::ExcessBalance)
```

Comment:
nits: Should this error be recorded to `error_counters` like others, or not?

Reply:
Ditto!